### PR TITLE
[receiver/hostmetrics] Add optional system.linux.memory.available metric

### PR DESCRIPTION
**Pull request by ItsLastDay**
**Description:** Implement a new metric `system.linux.memory.available`, which I defined in https://github.com/open-telemetry/semantic-conventions/issues/257.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7417

**Testing:** added a check to an existing unit test. I had to refactor the test a bit:  it assumed that a certain metric will be at position 0, which is not true now.

**Documentation:** Added note in `metadata.yaml`

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/memoryscraper/metadata.yaml
Comment (crobert-1) [inline receiver/hostmetricsreceiver/internal/scraper/memoryscraper/metadata.yaml:34]:
```suggestion
    description: An estimate of how much memory is available for starting new applications, without swapping. This is a more accurate alternative than system.memory.usage with state=free. (Linux only)
```
It might be good to have the actual description of what this metric is here, instead of simply a comparison to a different metric. I realize this might be a bit verbose so feel free to modify. Just a suggestion.

If it's obvious to users what this is for, feel free to ignore.

**Review by crobert-1 — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Comment by ItsLastDay** [comment]
Hi folks, gentle ping on this one :) IIUC I'm only missing one more LGTM - whom should I ask for that?

**Comment by crobert-1** [comment]
@dmitryax should review as code owner when possible.

**Review by dmitryax — APPROVED** [review]
@braydonk please review as well

**Review by braydonk — APPROVED** [review]
LGTM

**Review by atoulme — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**
